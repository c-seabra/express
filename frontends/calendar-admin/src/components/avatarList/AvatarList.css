/* RSVPs */
.avatarList {
  padding: 0;
  margin: 0;
  list-style: none;
  width: 100%;
}

.avatarList li {
  margin-bottom: 1rem;
}

/* END RSVPs */
