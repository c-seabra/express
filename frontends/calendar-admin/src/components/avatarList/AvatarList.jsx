import React from 'react'

import Avatar from '../avatar/Avatar'
import styles from './AvatarList.css'

const AvatarList = ({ iconActive, listType, iconClickCallback, avatarList, organizerId }) => (
  avatarList
    ? <ul className={styles.avatarList}>
      {avatarList.map(({ invitation, attendance }) => {
        return <li key={`${listType}_${invitation.id}`}>

          <Avatar
            src={attendance.data.person.avatar_url}
            alt='Invitee avatar'
            title={attendance.data.person.first_name + ' ' + attendance.data.person.last_name}
            response={invitation.response.response_status}
            invitationId={invitation.id}
            organizer={organizerId === (invitation.invitee && invitation.invitee.id)}
            {...{
              iconActive,
              listType,
              iconClickCallback
            }}
          />

        </li>
      }

      )}
    </ul>
    : null
)

export default AvatarList
