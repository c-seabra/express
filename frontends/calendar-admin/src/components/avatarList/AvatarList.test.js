import React from 'react';
import renderer from 'react-test-renderer';
import AvatarList from './AvatarList.jsx';

describe('AvatarList', () => {
    let realUseContext;
    let useContextMock;

    beforeEach(() => {
        realUseContext = React.useContext;
        React.useContext = jest.fn()
        useContextMock = React.useContext;
    });

    afterEach(() => {
        React.useContext = realUseContext;
    });

    test('renders without props', () => {
        useContextMock.mockReturnValue({});
        const component = renderer.create(
            <AvatarList />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('renders with props', () => {
        useContextMock.mockReturnValue({});
        const component = renderer.create(
            <AvatarList
                styles="styles"

                avatarList={[
                    {
                        invitation: { id: 123, response: { response_status: 'accepted' } },
                        attendance: { data: { person: { first_name: 'Tomasz' } } }

                    }
                ]}
                organizerId="123456"

                iconActive={true}
                iconType="delete"
                iconClickCallback={function () { return null }}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

});
