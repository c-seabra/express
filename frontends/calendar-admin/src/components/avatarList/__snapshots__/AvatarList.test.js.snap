// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarList renders with props 1`] = `
<ul
  className={undefined}
>
  <li>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <object
          data={undefined}
          type="image/png"
        >
          <img
            alt="Invitee avatar"
            src="https://s3-eu-west-1.amazonaws.com/assets.cilabs.com/images/person.svg"
            style={
              Object {
                "borderColor": "transparent",
              }
            }
          />
        </object>
        <svg
          viewBox="0 0 10 9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clipPath
              id="vrmva"
            >
              <path
                d="M4.125 5.725L2.9 4.5l-.408.408 1.633 1.634 3.5-3.5-.408-.409z"
              />
            </clipPath>
          </defs>
          <g>
            <g>
              <g>
                <path
                  d="M5 9a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                  fill="#f8992c"
                />
              </g>
              <g>
                <g />
                <g
                  clipPath="url(#vrmva)"
                >
                  <g>
                    <path
                      d="M1.5 1h7v7h-7z"
                      fill="#fff"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className={undefined}
      >
        <p
          className={undefined}
        >
          Tomasz undefined
        </p>
      </div>
      <span
        className={undefined}
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 10 9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clipPath
              id="en5za"
            >
              <path
                d="M7.333 2.637l-.47-.47L5 4.03 3.137 2.167l-.47.47L4.53 4.5 2.667 6.363l.47.47L5 4.97l1.863 1.863.47-.47L5.47 4.5z"
              />
            </clipPath>
          </defs>
          <g>
            <g>
              <g>
                <path
                  d="M4.95 9a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                  fill="#ed1846"
                />
              </g>
            </g>
            <g>
              <g />
              <g
                clipPath="url(#en5za)"
              >
                <g>
                  <path
                    d="M1 .5h8v8H1z"
                    fill="#fff"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </span>
    </div>
  </li>
</ul>
`;

exports[`AvatarList renders without props 1`] = `null`;
